decidim-core/lib/decidim/map/dynamic_map.rb

```diff
@@ -81,7 +81,7 @@ def stylesheet_snippets
 
         # @see Decidim::Map::View::Builder#javascript_snippets
         def javascript_snippets
-          template.javascript_pack_tag("decidim_map_provider_default", defer: false)
+          template.append_javascript_pack_tag("decidim_map_provider_default")
```

Comment:
We need to change this convention altogether if we are calling `template.append_javascript_pack_tag` here.

The idea of this method traditionally was that it generates the snippets we need to add to the view and the addition of these snippets is handled elsewhere.

For the maps case this is happening here where the snippets are added as part of the snippets to be printed in the view:
https://github.com/decidim/decidim/blob/03f9f993a2962d649281c15405687184beb3cf6d/decidim-core/app/helpers/decidim/map_helper.rb#L53-L55

And then those snippets are printed to the view here:
https://github.com/decidim/decidim/blob/03f9f993a2962d649281c15405687184beb3cf6d/decidim-core/app/views/layouts/decidim/_head.html.erb#L35

https://github.com/decidim/decidim/blob/03f9f993a2962d649281c15405687184beb3cf6d/decidim-core/app/views/layouts/decidim/_decidim_javascript.html.erb#L9

The `append_javascript_pack_tag` helper is not creating the snippet to be displayed in the view, so this convention would not apply when we are calling `template.append_javascript_pack_tag` here.

Another thing we need to check here is that the snippet is not appended twice in case the same page displays two separate maps.

Reply:
You are right.

Reply:
From what I can see in the shakapcker codebase, the javascript should not be added several times ... 
from what i have read in the [codebase](https://github.com/shakacode/shakapacker/blob/4c92e2a956c443233a6f00b65df7db67b5cfb2ba/lib/webpacker/helper.rb) 

`append_javascript_pack_tag` internally calls `javascript_pack_tag_queue` which creates entries in : `javascript_pack_tag_queue` split in defered and not defered. 

After that, `javascript_pack_tag` is calling internally `sources_from_manifest_entrypoints` with `javascript_pack_tag_queue`, which returns the unique names for assets being added. 

Also, the shakapacker is also doing a defered - not_defered, yielding 2 different arrays. 

https://github.com/shakacode/shakapacker/blob/4c92e2a956c443233a6f00b65df7db67b5cfb2ba/lib/webpacker/helper.rb#L98